.github/workflows/code-validation.yaml

```
@@ -27,6 +27,16 @@ jobs:
       if: matrix.test == 'validate-helm'
       uses: azure/setup-helm@v4.3.0
 
+    - name: Set up Python and install pre-commit (for precommit test)
+      if: matrix.test == 'precommit'
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.x'
+
+    - name: Install pre-commit
+      if: matrix.test == 'precommit'
+      run: pip install pre-commit
+
```

Since you now have a `requirements.txt` file for the doc stuff, might as well use it. This will give you the ability to use the `cache` feature.
```suggestion
    - name: Set up Python and install pre-commit (for precommit test)
      if: matrix.test == 'precommit'
      uses: actions/setup-python@v5
      with:
        python-version: '3.x'
        cache: pip

    - name: Install pre-commit
      if: matrix.test == 'precommit'
      run: pip install -r requirements.txt
```

And would argue to even remove the `if: matrix.test == 'precommit'` cause when we use the cache along with pip install, it works pretty fast. This is just because I personally try to avoid as much as possible having `if` in the CI. I like them dumb with the least amount of logic as possible, and leave that logic inside actual code.

---

Done, took your commit and updated requirements.txt with your caching step as well
Also removed if conditions. Personally, I don't like installing things when not needed, but I like your idea of least amount of logic as possible, and leave that logic inside actual code. TBH, I am going your way cuz I trust your opinion :)

---

Yeah I agree with installing the least amount of things, that is usually the way to go, but since we're not shipping any of this, and as long as it's stable and doesn't take too much of our time I think it's fine to make exceptions.